tensorflow/python/ops/signal/fft_ops.py

```diff
@@ -399,11 +399,11 @@ def ifftshift(x, axes=None, name=None):
     x = _ops.convert_to_tensor(x)
     if axes is None:
       axes = tuple(range(x.shape.ndims))
-      shift = [-int(dim // 2) for dim in x.shape]
+      shift = [-_math_ops.cast(dim // 2, _dtypes.int32) for dim in _array_ops.shape(x)]
     elif isinstance(axes, int):
-      shift = -int(x.shape[axes] // 2)
+      shift = -_math_ops.cast(_array_ops.shape(x)[axes] // 2, _dtypes.int32)
```

Comment:
The cast is unnecessary, though we should keep the order of operations (divide them negate):
```
shift = -(_array_ops.shape(x)[axes] // 2)
```

Reply:
yes